global_planner/src/planner_core.cpp

```diff
@@ -222,6 +227,15 @@ bool GlobalPlanner::makePlan(const geometry_msgs::PoseStamped& start, const geom
 bool GlobalPlanner::makePlan(const geometry_msgs::PoseStamped& start, const geometry_msgs::PoseStamped& goal,
                            double tolerance, std::vector<geometry_msgs::PoseStamped>& plan) {
     boost::mutex::scoped_lock lock(mutex_);
+
+    // reset last_official_goal if it diverges
+    if (fabs(last_official_goal_x - goal.pose.position.x) > 0.1 || fabs(last_official_goal_y - goal.pose.position.y) > 0.1) {
+      last_official_goal_x = goal.pose.position.x;
+      last_official_goal_y = goal.pose.position.y;
+      new_goal_x = goal.pose.position.x;
+      new_goal_y = goal.pose.position.y;
+      ROS_DEBUG("NEW GOAL, setting last goal to: %f %f", new_goal_x, new_goal_y);
+    }
```

Comment:
If `initialize` is not called, then `last_official_goal_x` will not be initialized, so either, you should
 * Move the initialization to the constructor's initializers (preferred)
 * or Move this if statement to after the initialized check on the next line.